tutorials/core_components.qmd

```diff
@@ -0,0 +1,344 @@
+engine: julia
+---
+
+BiochemicalAlgorithms.jl`s design is inspired by its predecessor [```BALL```](https://github.com/ball-project/ball). BALL and BiochemicalAlgorithms.jl both have a set of KERNEL classes or core classes to represent molecular entities such as Molecules, Proteins, Atoms, Bonds etc. The following class diagramm shows the different classes and their relationship to each others.
+
+![UML class diagram *Note*: Only the most important functionalities are shown here.](uml.png)
+
+In the center of all classes resides the [`System`](@ref). So let's see it in action with a simple peptide:
+```{julia}
+using BiochemicalAlgorithms
+
+s = load_pdb(ball_data_path("../test/data/AlaAla.pdb"))
+s
+```
+
+This will give us an overview of the objects associated with our system. It contains 23 atoms, 0 bonds, 1 molecule, 1 chain, 0 secondary structures and 2 fragments. We look at each group separately in the following sections.
+
+If we do not explicitely create a system, the data will be stored in the default system:
+```{julia}
+load_pdb(ball_data_path("../test/data/AlaAla.pdb"))
+```
+
+
+## Atom
+
+We can access the atoms like this:
+```{julia}
+sys_atms = atoms(s)
+
+natoms(sys_atms)
+```
+
+Let's play around with it:
+```{julia}
+println("Atom elements:")
+for a in sys_atms
+    print(a.element)
+end
+println()
+
+n1= atom_by_name(s, "N")
+
+println(n1.idx)
+
+```
+
+We can access an atom by its name, which will return the first atom matching the given name. If there are more atoms with the same name and we are not interested in the first one, we can access the atom by the atom index.
+The atom index is unique inside a system but not necessarily starting by one.
+
+```{julia}
+println("Atom ids:")
+for a in sys_atms
+    println(a.idx, " ", a.element)
+end
+
+n2 = atom_by_idx(s, 17)
+
+```
+
+[`Atom`](@ref) features the atom number as well. The atom number does NOT have to be unique inside the system!
+
+```{julia}
+for a in sys_atms
+    println(a.idx, " ", a.number)
+end
+```
+
+Next, we will have a look at `Bond`.
+
+## Bond
+
+BiochemicalAlgorithms.jl supports different BondTypes:
+```{julia}
+BiochemicalAlgorithms.BondOrderType
+```
+
+Typically, bonds are not included in the PDB-FileFormat and have to be computed. This can be achieved with the FragmentDataBase, which contains known fragments (all amino acids such as alanine, all nucleotides, same ions ...). If you are interested in the fragments contained in the default FragmentDB, take a look at data/fragments of the repository. There you find all the fragments, each one stored in a separate json format.
+So let's take a look at the FragmentDB and bonds:
+```{julia}
+s = load_pdb(ball_data_path("../test/data/AlaAla.pdb"))
+nbonds(s)
+```
+
+So far, the system has no bonds. We can also visualize the structure using BiochemicalVisualization.jl.
+
+If BiochemicalVisualization is not yet installed, you can add it via:
+
+```{julia}
+using Pkg
+Pkg.add("BiochemicalVisualization")
+```
```

Comment:
````suggestion
```{julia}
#| eval: false

using Pkg
Pkg.add("BiochemicalVisualization")
```
````
This should not be executed!